src/walk.rs

```diff
@@ -36,13 +36,99 @@ enum ReceiverMode {
 
 /// The Worker threads can result in a valid entry having PathBuf or an error.
 #[allow(clippy::large_enum_variant)]
+#[derive(Debug)]
 pub enum WorkerResult {
     // Errors should be rare, so it's probably better to allow large_enum_variant than
     // to box the Entry variant
     Entry(DirEntry),
     Error(ignore::Error),
 }
 
+/// A batch of items to send over a channel.
+struct Batch<T> {
+    items: Arc<Mutex<Option<Vec<T>>>>,
+}
+
+impl<T> Batch<T> {
+    fn new() -> Self {
+        Self {
+            items: Arc::new(Mutex::new(Some(vec![]))),
+        }
+    }
+
+    fn lock(&self) -> MutexGuard<'_, Option<Vec<T>>> {
+        self.items.lock().unwrap()
+    }
+}
+
+impl<T> Clone for Batch<T> {
```

Comment:
I believe this implementation can be derived.

Reply:
`derive(Clone)` would have added a `T: Clone` bound not satisfied by `WorkerResult`.  But I just switched it to being non-generic so it can derive it now.